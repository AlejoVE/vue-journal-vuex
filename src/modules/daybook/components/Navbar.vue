<template>
    <nav class="navbar bg-primary">
        <a class="navbar-brand text-white">
            <img class="d-inline-block align-text-top mx-2" src="@/assets/logo.png" height="24" alt="logo"/>
            {{ username }}
        </a>
        <div class="d-flex">
            <button @click="onLogout" class="btn btn-outline-info mx-5 my-2"><i class="fa fa-sign-out-alt"></i></button>
        </div>
    </nav>
</template>

<script>
import { useRouter } from 'vue-router'
import useAuth from '../../auth/composables/useAuth'
export default {
    name: 'NavBar',
    setup(){

        const router = useRouter()

        const { logout, username } = useAuth()

        return {
            username,
            onLogout(){
                router.push({ name: 'login' })
                logout()
            }
        }

    }
}
</script>