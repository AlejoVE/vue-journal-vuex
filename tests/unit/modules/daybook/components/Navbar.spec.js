import { shallowMount } from '@vue/test-utils'
import Navbar from '@/modules/daybook/components/Navbar.vue'

 import createVuexStore from "../../../mocks-data/mock-store"


describe('Tests in Navbar component', () => {

    const store = createVuexStore({
        user: {
            name: 'Luis',
            email: 'luis@gmail.com'
        },
        status: 'authenticated',
        idToken: 'ABC',
        refreshToken: 'XYZ'
    })

    beforeEach(() => jest.clearAllMocks())

    test('should show component correctly', () => {
      const wrapper = shallowMount( Navbar, {
          global: {
              plugins: [store]
          }
      })

     expect(wrapper.html()).toMatchSnapshot()
    })

    test('should logout the user and redirect to login',  async () => {
        const wrapper = shallowMount( Navbar, {
            global: {
                plugins: [store]
            }
        })

        await wrapper.find('button').trigger('click')

        expect(wrapper.router.push).toHaveBeenCalledWith({ name: 'login' })

        expect(store.state.auth).toEqual({
            user: null,
            status: 'not-authenticated',
            idToken: null,
            refreshToken: null
        })
    })
    
    
})
