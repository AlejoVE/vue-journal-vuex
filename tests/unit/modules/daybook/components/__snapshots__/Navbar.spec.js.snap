// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Navbar component should show component correctly 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white"><img class="d-inline-block align-text-top mx-2" src="" height="24" alt="logo"> Luis</a>
  <div class="d-flex"><button class="btn btn-outline-info mx-5 my-2"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
